### PR TITLE
Select and copy

### PR DESCRIPTION
**Pull request by shimondoodkin**
I have updated to pyqt5 I hope it is ok

### CODE REVIEW

File: termqt/terminal_widget.py [hunk 1]
```diff
@@ -210,23 +264,28 @@ def _paint_buffer(self):
             if real_ln < 0 or real_ln >= len(self._buffer):
                 break
 
-            row = self._buffer[ln + offset]
+            row = self._buffer[real_ln]
 
             ht += lh
             for cn, c in enumerate(row):
                 if c:
+                    is_selected = self._is_selected(cn, real_ln)
```
Comment (TerryGeng) [inline termqt/terminal_widget.py:272]: For the sake of executing speed, I would be reluctant to place a method call inside the paint loop. Given that the selection region is sequential, I think this part of the code can be optimized and put inside the loop.

**Comment by TerryGeng** [comment]
I think it is easier for me to clean up this PR first and then proceed to disseminate new changes from your new PR.

I tried to push to your branch `select-and-copy` but my push was rejected. Do you mind granting me write permission to this branch? Or I can also start my own branch and a different PR if you don't feel comfortable doing so.

**Comment by TerryGeng** [comment]
Ok. Thanks for the write permission. I reformatted some commit messages and squashed some commits.

I also PyQt5 back to Qt. Please checkout #8 for more details.

Do they look good to you?

**Comment by shimondoodkin** [comment]
> Ok. Thanks for the write permission. I reformatted some commit messages and squashed some commits.
> 
> I also changed PyQt5 back to Qt. Please check out #8 for more details.
> 
> Do they look good to you?



> Ok. Thanks for the write permission. I reformatted some commit messages and squashed some commits.
> 
> I also changed PyQt5 back to Qt. Please check out #8 for more details.
> 
> Do they look good to you?

I have not tested.
it looks ok

I was just worried about reverting the qt5 to qt that there was a different name for Signal 
I looked at it in code and it was fine.

**Comment by TerryGeng** [comment]
Ok. I will proceed to merge. Again, thanks for your contribution!

**Comment by TerryGeng** [comment]
Actually, I was working on some optimization while waiting for your feedback. I pushed it to the master branch just now mainly because I really want to get the utf-8 issue to be fixed asap.

Now I think about it, I probably should have submitted this commit to this PR and waited for your feedback...

I did some tests and it doesn't cause regression to my knowledge. I wish you may give it a quick look if you have a chance.

Regarding the optimization of the paint logic, my idea is to reduce branching while keeping the code duplication manageable. However, branching is not too bad an issue given modern processors have very decent branch prediction.

**Pull request merged**